native/src/seal/decryptor.cpp

```diff
@@ -314,39 +314,99 @@ namespace seal
+                {
+                    throw invalid_argument("tile_size does not divide coeff_count");
+                }
+#endif
 
-        // Compute dyadic product with secret power array
-        auto secret_key_array = PolyIter(secret_key_array_.get(), coeff_count, key_coeff_modulus_size);
-        SEAL_ITERATE(iter(encrypted_copy, secret_key_array), encrypted_size - 1, [&](auto I) {
-            dyadic_product_coeffmod(get<0>(I), get<1>(I), coeff_modulus_size, coeff_modulus, get<0>(I));
-        });
+                // Semantic misuse of RNSIter; each is really pointing to the data for each RNS factor in
+                // sequence
+                ConstRNSIter c0_tile(encrypted.data(0), tile_size);
+                ConstRNSIter c1_tile(encrypted.data(1), tile_size);
+                ConstRNSIter sk_tile(secret_key_array_.get(), tile_size);
+                RNSIter dest_tile(destination, tile_size);
+                SEAL_ITERATE(iter(coeff_modulus, ntt_tables), coeff_modulus_size, [&](auto I) {
+                    SEAL_ITERATE(iter(size_t(0)), num_tiles, [&](SEAL_MAYBE_UNUSED auto J) {
+                        // put < c_1 * s > mod q in destination
+                        dyadic_product_coeffmod(c1_tile[0], sk_tile[0], tile_size, get<0>(I), dest_tile[0]);
+                        // Add c_0 to the result
+                        add_poly_coeffmod(dest_tile[0], c0_tile[0], tile_size, get<0>(I), dest_tile[0]);
+
+                        // Manually increment iterators
+                        ++c0_tile;
+                        ++c1_tile;
+                        ++sk_tile;
+                        ++dest_tile;
+                    });
+                });
+            }
+            else
+            {
+                // No tiling, since the NTT isn't easily tiled
+                ConstRNSIter secret_key_array(secret_key_array_.get(), coeff_count);
+                ConstRNSIter c0(encrypted.data(0), coeff_count);
+                ConstRNSIter c1(encrypted.data(1), coeff_count);
+                SEAL_ITERATE(
+                    iter(c0, c1, secret_key_array, coeff_modulus, ntt_tables, destination), coeff_modulus_size,
+                    [&](auto I) {
+                        set_uint(get<1>(I), coeff_count, get<5>(I));
+                        // Transform c_1 to NTT form
+                        ntt_negacyclic_harvey_lazy(get<5>(I), get<4>(I));
+                        // put < c_1 * s > mod q in destination
+                        dyadic_product_coeffmod(get<5>(I), get<2>(I), coeff_count, get<3>(I), get<5>(I));
+                        // Transform back
+                        inverse_ntt_negacyclic_harvey(get<5>(I), get<4>(I));
+                        // add c_0 to the result; note that destination should be in the same (NTT) form as encrypted
+                        add_poly_coeffmod(get<5>(I), get<0>(I), coeff_count, get<3>(I), get<5>(I));
+                    });
+            }
+        } else {
```

Comment:
Could you break this into 3 lines? It's weird that pre-commit did not capture this.

Reply:
Done; running `pre-commit run --all-files` shows many lines whose format doesn't comply to the format, so I've skipped the pre-commit step for the current PR.